Library/Homebrew/formula_cellar_checks.rb

```diff
@@ -315,58 +315,6 @@ def check_cpuid_instruction(formula)
     "No `cpuid` instruction detected. #{formula} should not use `ENV.runtime_cpu_detection`."
   end
 
-  def check_binary_arches(formula)
```

Comment:
I think ideally rather than duplicate this in both the `extend` files it would be good to have something like `def actually_check_binary_arches` which keeps the existing `def check_binary_arches` implementation and make the `def check_binary_arches` in this file be empty/a no-op and in the two `extend` files call `actually_check_binary_arches`. 

Does that make sense? 

Thanks again @hyuraku!

Reply:
@MikeMcQuaid 
Thank you for the advice.
Let me know if the implementation you mentioned is the following or not.

Library/Homebrew/formula_cellar_checks.rb
```rb
def check_binary_arches(formula)
    # There is no `binary_executable_or_library_files` method for the generic OS
end

def actually_check_binary_arches(formula)
   return unless formula.prefix.directory?

  --< omitted >--
end
```

Library/Homebrew/extend/os/mac/formula_cellar_checks.rb
```rb
def check_binary_arches(formula)
   actually_check_binary_arches(formula)
end
```

Reply:
Yup that sounds right and the same thing as `os/mac` in `os/linux` too, thanks @hyuraku!

Reply:
Perhaps a different (and possibly cleaner) solution here is to implement a `binary_executable_or_library_files` method for the generic OS that returns an empty array, and we can have the specific implementations for Mac and Linux in the right places.

Reply:
So the diff would look something like this:
```diff
diff --git a/Library/Homebrew/extend/os/keg.rb b/Library/Homebrew/extend/os/keg.rb
index 5e3cdd073..733768bff 100644
--- a/Library/Homebrew/extend/os/keg.rb
+++ b/Library/Homebrew/extend/os/keg.rb
@@ -1,4 +1,8 @@
 # typed: strict
 # frozen_string_literal: true
 
-require "extend/os/mac/keg" if OS.mac?
+if OS.mac?
+  require "extend/os/mac/keg"
+elsif OS.linux?
+  require "extend/os/linux/keg"
+end
diff --git a/Library/Homebrew/keg.rb b/Library/Homebrew/keg.rb
index a3a5579c5..cb19e3af1 100644
--- a/Library/Homebrew/keg.rb
+++ b/Library/Homebrew/keg.rb
@@ -530,7 +530,7 @@ class Keg
   end
 
   def binary_executable_or_library_files
-    elf_files
+    []
   end
 
   def codesign_patched_binary(file); end
```
And we would create an `extend/os/linux/keg.rb` containing
```ruby
# typed: false
# frozen_string_literal: true

class Keg
  undef binary_executable_or_library_files

  def binary_executable_or_library_files
    elf_files
  end
end
```

Reply:
> Perhaps a different (and possibly cleaner) solution here is to implement a `binary_executable_or_library_files` method for the generic OS that returns an empty array, and we can have the specific implementations for Mac and Linux in the right places.

Yes, that sounds even nicer!